planner/core/casetest/testdata/derive_topn_from_window_out.json

```diff
@@ -196,18 +331,98 @@
+          "        └─TopN 1.00 cop[tikv]  test.t.a, offset:0, count:1",
+          "          └─TableFullScan 10000.00 cop[tikv] table:t keep order:false, stats:pseudo"
+        ],
+        "Res": [
+          "1"
+        ]
+      },
+      {
+        "SQL": "select * from (select row_number() over (partition by b) as rownumber from t where a >= 2) DT where rownumber <= 3 order by rownumber  -- pattern is applicable with partition by prefix and filter on data source",
+        "Plan": [
+          "Sort 3.00 root  Column#4",
+          "└─Projection 3.00 root  Column#4",
+          "  └─Shuffle 3.00 root  execution info: concurrency:3, data sources:[Limit]",
+          "    └─Window 3.00 root  row_number()->Column#4 over(partition by test.t.b rows between current row and current row)",
+          "      └─Sort 3.00 root  test.t.b",
+          "        └─Limit 3.00 root  offset:0, count:3",
+          "          └─TableReader 3.00 root  data:Limit",
+          "            └─Limit 3.00 cop[tikv]  offset:0, count:3",
+          "              └─Selection 3.00 cop[tikv]  ge(test.t.a, 2)",
+          "                └─TableFullScan 9.00 cop[tikv] table:t keep order:false, stats:pseudo"
+        ],
+        "Res": [
+          "1",
+          "1",
+          "2"
+        ]
+      },
+      {
+        "SQL": "select * from (select row_number() over () as rownumber from tt) DT where rownumber <= 3 -- pattern is applicable with no partition on a non-clustered table",
+        "Plan": [
+          "Projection 3.00 root  Column#6",
+          "└─Window 3.00 root  row_number()->Column#6 over(rows between current row and current row)",
+          "  └─Limit 3.00 root  offset:0, count:3",
+          "    └─IndexReader 3.00 root  index:Limit",
+          "      └─Limit 3.00 cop[tikv]  offset:0, count:3",
+          "        └─IndexFullScan 3.00 cop[tikv] table:tt, index:PRIMARY(b, a) keep order:false, stats:pseudo"
+        ],
+        "Res": [
+          "1",
+          "2",
+          "3"
+        ]
+      },
+      {
+        "SQL": "select * from (select *, row_number() over (partition by primary_key, secondary_key order by c_timestamp) as rownum from customer where primary_key = 0x002 and secondary_key >= 0x001 and c_timestamp >= 1661883508511000000) as nested where rownum <= 10 order by secondary_key desc;",
+        "Plan": [
+          "Sort 1.11 root  test.customer.secondary_key:desc",
+          "└─Window 1.11 root  row_number()->Column#6 over(partition by test.customer.primary_key, test.customer.secondary_key order by test.customer.c_timestamp rows between current row and current row)",
+          "  └─Sort 1.11 root  test.customer.primary_key, test.customer.secondary_key, test.customer.c_timestamp",
+          "    └─TopN 1.11 root  test.customer.c_timestamp, offset:0, count:10",
+          "      └─TableReader 1.11 root  data:TopN",
+          "        └─TopN 1.11 cop[tikv]  test.customer.c_timestamp, offset:0, count:10",
```

Comment:
can we add partition by column to explain the result?
like "TopN 1.11 cop[tikv]  order_by:XXX desc, partition_by: XXX, offset: X, count: X"

Reply:
Good point and will do in next update.